make/post/rules/help.mak

```diff
@@ -20,12 +21,77 @@
 #      targets.
 #
 
-define PrintHelp
+define PrintHelpPreamble
 $(NL_V_AT)echo "This makefile supports the following build targets:"
 $(NL_V_AT)echo
+endef # PrintHelpPreamble
+
+define PrintAllHelp
+$(NL_V_AT)echo "  all"
+$(NL_V_AT)echo "    Generate all configured build artifacts for this project."
+$(NL_V_AT)echo
+endef # PrintAllHelp
+
+define PrintCheckHelp
+$(NL_V_AT)echo "  check"
+$(NL_V_AT)echo "    Generate all configured build artifacts and run all unit"
+$(NL_V_AT)echo "    and functional tests for this project."
+$(NL_V_AT)echo
+endef # PrintCheckHelp
+
+define PrintDistHelp
+$(NL_V_AT)echo "  dist"
+$(NL_V_AT)echo "    Generate an archive distribution snapshot for this project."
+$(NL_V_AT)echo
+endef # PrintDistHelp
+
+define PrintDistcheckHelp
+$(NL_V_AT)echo "  distcheck"
+$(NL_V_AT)echo "    Generate an archive distribution snapshot for this project"
+$(NL_V_AT)echo "    and sanity check the resulting distribution by running"
+$(NL_V_AT)echo "    'make check' on it for this project."
+$(NL_V_AT)echo
+endef # PrintDistcheckHelp
+
+define PrintDocHelp
+$(NL_V_AT)echo "  doc"
+$(NL_V_AT)echo "    Generate documentation for the project."
+$(NL_V_AT)echo
+endef # PrintDocHelp
+
+define PrintDocdistHelp
+$(NL_V_AT)echo "  docdist"
+$(NL_V_AT)echo "    Generate an archive distribution of the documentation for"
+$(NL_V_AT)echo "    the project."
+$(NL_V_AT)echo
+endef # PrintDocdistHelp
+
+define PrintHelpHelp
 $(NL_V_AT)echo "  help"
 $(NL_V_AT)echo "    Display this content."
 $(NL_V_AT)echo
+endef # PrintHelpHelp
+
+define PrintInstallHelp
+$(NL_V_AT)echo "  install"
+$(NL_V_AT)echo "    Generate all configured build artifacts for this project"
+$(NL_V_AT)echo "    and install athem in DESTDIR on the build host system."
+$(NL_V_AT)echo
+endef # PrintInstallHelp
+
+define PrintHelp
+$(call PrintHelpPreamble)
```

Comment:
no need to $(call ..) unless you need to bind or the function uses numbered args...